modulegen/internal/context/context.go

```diff
@@ -68,38 +68,38 @@ func (ctx *Context) getMarkdownsFromDir(baseDir string) ([]string, error) {
 	return dirs, nil
 }
 
-func (ctx *Context) GetExamples() ([]string, error) {
+func (ctx Context) GetExamples() ([]string, error) {
 	return ctx.getModulesByBaseDir("examples")
 }
 
-func (ctx *Context) GetModules() ([]string, error) {
+func (ctx Context) GetModules() ([]string, error) {
 	return ctx.getModulesByBaseDir("modules")
 }
 
-func (ctx *Context) GetExamplesDocs() ([]string, error) {
+func (ctx Context) GetExamplesDocs() ([]string, error) {
 	return ctx.getMarkdownsFromDir("examples")
 }
 
-func (ctx *Context) GetModulesDocs() ([]string, error) {
+func (ctx Context) GetModulesDocs() ([]string, error) {
 	return ctx.getMarkdownsFromDir("modules")
 }
 
-func (ctx *Context) MkdocsConfigFile() string {
+func (ctx Context) MkdocsConfigFile() string {
 	return filepath.Join(ctx.RootDir, "mkdocs.yml")
 }
 
-func (ctx *Context) VSCodeWorkspaceFile() string {
+func (ctx Context) VSCodeWorkspaceFile() string {
 	return filepath.Join(ctx.RootDir, ".vscode", ".testcontainers-go.code-workspace")
 }
 
-func New(dir string) *Context {
-	return &Context{RootDir: dir}
+func New(dir string) Context {
+	return Context{RootDir: dir}
 }
 
-func GetRootContext() (*Context, error) {
+func GetRootContext() (Context, error) {
 	current, err := os.Getwd()
 	if err != nil {
-		return nil, err
+		return Context{}, err
```

Comment:
Why not return a pointer ?

Reply:
I'm removing the pointer semantics as we are not in the need of modifying the content of the struct

Reply:
OK